std/asyncio/filesystem/FileSystem.hx

```diff
@@ -0,0 +1,279 @@
+		callback.fail(new NotImplemented());
+	}
+
+	/**
+		Create a unique temporary directory.
+
+		For a directory name `prefix` gets appended with random characters.
+		The created directory path is passed to the `callback`.
+
+		Created directory will _not_ be deleted automatically.
+	**/
+	static public function createTempDirectory(prefix:FilePath, callback:Callback<Null<FilePath>>):Void {
+		callback.fail(new NotImplemented());
+	}
+
+	/**
+		Renames the file or directory located at `oldPath` to `newPath`.
+
+		If `newPath` already exists and `overwrite` is `true` (which is the default)
+		the destination is overwritten.
+	**/
+	static public function rename(oldPath:FilePath, newPath:FilePath, overwrite:Bool = true, callback:Callback<NoData>):Void {
+		callback.fail(new NotImplemented());
+	}
+
+	/**
+		Remove a file or symbolic link.
+	**/
+	static public function deleteFile(path:FilePath, callback:Callback<NoData>):Void {
+		callback.fail(new NotImplemented());
+	}
+
+	/**
+		Remove an empty directory.
+	**/
+	static public function deleteDirectory(path:FilePath, callback:Callback<NoData>):Void {
+		callback.fail(new NotImplemented());
+	}
+
+	/**
+		Recursively remove everything at the given `path`.
+
+		Removes files, symbolic links and recursively removes directories and their contents.
+	**/
+	static public function deleteRecursive(path:FilePath, callback:Callback<NoData>):Void {
+		callback.fail(new NotImplemented());
+	}
+
+	/**
+		Get file or directory information at the given path.
+	**/
+	static public function info(path:FilePath, callback:Callback<Null<FileInfo>>):Void {
+		callback.fail(new NotImplemented());
+	}
+
+	/**
+		Check user's access for a path.
+
+		Example:
+		```haxe
+		import asyncio.filesystem.FileAccessMode;
+		//check path existence
+		FileSystem.check(path, Exists, (error, result) -> trace(result));
+		//check if file is executable
+		FileSystem.check(path, Executable, (error, result) -> trace(result));
+		//check if file is readable and writable
+		FileSystem.check(path, Readable | Writable, (error, result) -> trace(result));
+		```
+	**/
+	static public function check(path:FilePath, mode:FileAccessMode, callback:Callback<Bool>):Void {
+		callback.fail(new NotImplemented());
+	}
+
+	/**
+		Set path permissions.
+
+		If `recursive` is `true` and `path` points to a directory: apply `mode`
+		recursively to the directory contents as well.
+	**/
+	static public function setPermissions(path:FilePath, mode:FileAccessMode, recursive:Bool = false, callback:Callback<NoData>):Void {
+		callback.fail(new NotImplemented());
+	}
+
+	/**
+		Set path owner.
+
+		If `recursive` is `true` and `path` points to a directory: apply recursively
+		to the directory contents as well.
+	**/
+	static public function setOwner(path:FilePath, user:SystemUser, ?group:SystemGroup, recursive:Bool = false, callback:Callback<NoData>):Void {
```

Comment:
Added a separate method `FileSystem.setGroup` because both `SystemUser` and `SystemGroup` has `from Int`. That makes argument skipping dangerous here.